client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -1416,11 +1417,13 @@ public void request(Message msg, final RequestCallback requestCallback, long tim
             @Override
             public void onSuccess(SendResult sendResult) {
                 requestResponseFuture.setSendRequestOk(true);
+                requestResponseFuture.executeRequestCallback();
             }
 
             @Override
             public void onException(Throwable e) {
                 requestResponseFuture.setCause(e);
+                requestResponseFuture.executeRequestCallback();
                 requestFail(correlationId);
```

Comment:
' requestResponseFuture.executeRequestCallback' has been called in  'requestFail(correlationId)'
you don't need to call it again.

Reply:
done